VstsTasks/AzureDtlCreateVM/task.ps1

```diff
@@ -117,14 +128,10 @@ try
             else
             {
                 Write-Host "A deployment failure occured. Retrying deployment (attempt $i of $($count - 1))"
-                if ($resourceId)
-                {
-                    Remove-AzureRmResource -ResourceId $resourceId -Force | Out-Null
-                }
-                else
+                Remove-FailedResourcesBeforeRetry -DeploymentName $deploymentName -ResourceGroupName $resourceGroupName -DeleteDeployment $DeleteFailedDeploymentBeforeRetry
+                if ($AppendRetryNumberToVMName)
                 {
-                    Write-Host "Resource identifier is not available, will not attempt to remove corresponding resouce before retrying."
-                    Write-Host "Dumping raw deployment result:`n$result"
+                    $templateParameterObject.newVMName = "$baseVmName-$i"

```

Comment:
Do we need to worry about VM name length at this point?

Reply:
We do have logic to check for that already. In the flag description we do mention that enabling this flag may increase and therefore fail VM name limits.